tests/compiler/std/set.optimized.wat

```diff
@@ -4251,14 +4250,57 @@
   i32.store offset=20
   local.get $0
  )
+ (func $~lib/set/Set<u16>#find (param $0 i32) (param $1 i32) (param $2 i32) (result i32)
```

Comment:
Wondering what happend here. @MaxGraey any ideas why this doesn't optimize away anymore?

Reply:
It just add one more monomorphized version for `u16`. I'm wondering could we rewrite `Set#find` in more sign agnostic way?

Reply:
Ah, I got why this happened. Before `sign-extension` `$~lib/set/Set<u16>#find` and `$~lib/set/Set<i16>#find` was pretty similar and folded (dedup) to one function, but with `sign-extension` they becomes looks pretty different and can't deduplicated anymore

Reply:
Ah, I see. Makes sense with the signed variant now utilizing sign extension while the unsigned variant does not. Too sad that it also doesn't optimize to the same code anymore.